pkg/services/baremetal/host/host.go

```diff
@@ -1437,6 +1409,26 @@ func (s *Service) actionEnsureProvisioned(_ context.Context) (ar actionResult) {
 		return actionContinue{delay: 10 * time.Second}
 	}
 
+	defer func() {
+		// Try to create an Event which contains the cloud-init-output.
+		errMsg := ""
+		switch v := ar.(type) {
+		case actionComplete:
+			return
+		case actionContinue:
+			return
+		case actionError:
+			if errors.Is(v.err, errGettingCloudInitOutput) {
+				return
+			}
+			errMsg = v.err.Error()
+		}
+		out := sshClient.GetCloudInitOutput()
+		record.Warnf(s.scope.HetznerBareMetalHost, "CloudInitOutput", "cloud init output (%s):\n%s",
```

Comment:
why do you use "warnf"?

and why do we use the defer statement at all btw? Why cannot we create it in the normal flow? The defer makes it harder to understand. Right now I also don't understand why we call the function once inside the defer and once outside

Reply:
> why do you use "warnf"?

I use Warnf() because I use formatting in the string (`%s`).

> and why do we use the defer statement at all btw? Why cannot we create it in the normal flow? The defer makes it harder to understand. Right now I also don't understand why we call the function once inside the defer and once outside

Good point. I did it like this, because it was done like that before.

I look if this can be done without defer.

Reply:
@janiskemper I updated the PR.